roles/kubernetes/kubeadm/tasks/main.yml

```diff
@@ -168,3 +169,11 @@
     - kubeadm_discovery_address != kube_apiserver_endpoint
   tags:
     - kube-proxy
+
+- name: Extract etcd certs from control plane if using etcd kubeadm mode
+  include_tasks: kubeadm_etcd_node.yml
+  when:
+    - etcd_mode == "kubeadm"
+    - kubeadm_control_plane
+    - inventory_hostname not in groups['kube-master']
+    - kube_network_plugin in ['calico', 'flannel', 'canal', 'cilium']
```

Comment:
Why only these network plugins?

it does not work with Kubernetes < 1.14 too. So this is a breaking change.

Reply:
This is a new feature. Existing clusters cannot switch to it. It's only for new clusters running v1.14 or newer. We're not changing the default value. The old mode is still the default.

Reply:
but why the list of network plugins ?

if this is a new feature, then we should add a pre check to verify settings in preinstall that if this is enabled and kube_version is less than 1.14 then fail

Reply:
@woopstar sure I'll add some checks to ensure some conditions, like v1.14.0 min version and kubeadm_control_plane is true.

This step is necessary so that kube-nodes get a client cert for etcd. They are only necessary for plugins that talk to etcd. The others don't need it, so we shouldn't run this step.

Reply:
I see. It should only run for Calico when the calico datastore is set to `etcd` (default) and not `kdd`

```
# calico_datastore: "etcd"
```